integration_tests/__init__.py

```diff
@@ -59,15 +60,18 @@ def setUp(self):
                 os.getcwd(), 'bin', 'snapcraft-parser')
 
         self.snaps_dir = os.path.join(os.path.dirname(__file__), 'snaps')
+
+        # share the cache in all tests.
+        self.useFixture(fixtures.EnvironmentVariable(
+            'XDG_CACHE_HOME', BaseDirectory.xdg_cache_home))
```

Comment:
It strikes me as wrong to use the actual cache home since it could hide bugs related to downloading missing packages. I'd expect a temporary folder to be used for all tests.

Reply:
The realization on my side is, the unit tests are fine with isolation. The integration tests and demos are the ones we need to fix excepting the integration tests that actually test the cache.